src/iface/interface/mod.rs

```diff
@@ -1337,8 +1337,9 @@ impl InterfaceInner {
             }
         }
 
-        if tcp_repr.control == TcpControl::Rst {
+        if tcp_repr.control == TcpControl::Rst || !ip_repr.dst_addr().is_unspecified() {
```

Comment:
I think the condition is reversed? if dst addr is unspecified you don't want to reply.

```suggestion
        if tcp_repr.control == TcpControl::Rst || ip_repr.dst_addr().is_unspecified() {
```

Reply:
Maybe we should just check for both src and dst. I don't think either should be unspecified.

Reply:
Yes indeed, it was inverted. Weird that CI did not catch it. I'll see if I can add a test for that.